ubuntu/entrypoint.sh

```diff
@@ -80,6 +84,84 @@ function unload_modules() {
     done
 }
 
+
+# find all available vfs
+# return format <pf_name> <vf_name> <vf_index> <vf_mac> <vf_pci_addr>
+function find_mlx_vfs() {
+    for ethpath in /sys/class/net/*; do
+        if (grep $VENDOR "$ethpath"/device/vendor >/dev/null 2>&1); then
+            for vf in "$ethpath"/device/virtfn[0-9]*/net/*; do
+                [[ -d $vf ]] && [[ -e $vf/address ]] && [[ -L $vf/device ]] || continue
+                pf_name=$(basename $ethpath) && [[ -n $pf_name ]] || return 1
+                vf_name=$(basename $vf) && [[ -n $vf_name ]] || return 1
+                vf_index=$(sed -E 's|.*/virtfn([0-9]+)/.*|\1|' <<<$vf) && [[ -n $vf_index ]] || return 1
+                vf_mac=$(cat $vf/address) && [[ -n $vf_mac ]] || return 1
+                vf_pci_addr=$(basename $(readlink $vf/device)) && [[ -n $vf_pci_addr ]] || return 1
+
+                echo "$pf_name" "$vf_name" "$vf_index" "$vf_mac" "$vf_pci_addr"
+            done
+        fi
+    done
+}
+
+function set_administrative_mac_for_vf() {
+    ip link set dev "$1" vf "$2" mac "$3"
```

Comment:
general comment it better to assign the $1 $2 .. to local variable for readability

Reply:
done